src/jit/macro.expr

```diff
@@ -0,0 +1,62 @@
+# -*-whitespace-*-
+
+(macro: ^getf (,object ,type ,field)
+   (load (addr ,object (&offsetof ,type ,field)) (&SIZEOF_MEMBER ,type ,field)))
+(macro: ^setf (,object ,type ,field ,value)
+    (store (addr ,object (&offsetof ,type ,field)) ,value (&SIZEOF_MEMBER ,type ,field)))
+(macro: ^cu_string (,a) (idx  (^getf (cu) MVMCompUnit body.strings) ,a ptr_sz))
+(macro: ^cu_callsite (,a) (idx  (^getf (cu) MVMCompUnit body.callsites) ,a ptr_sz))
+
+(macro: ^frame () (^getf (tc) MVMThreadContext cur_frame))
+(macro: ^parg (,a) (idx (^getf (^frame) MVMFrame params.args) ,a reg_sz))
+
+(macro: ^params () (addr (^frame) (&offsetof MVMFrame params)))
+(macro: ^caller () (addr (^frame) (&offsetof MVMFrame caller)))
+
+# get spesh slot address
+(macro: ^spesh_slot (,a)
+   (idx (^getf (^frame) MVMFrame effective_spesh_slots) ,a ptr_sz))
```

Comment:
Nothing to resolve right now, but I'm curious how we do the "if it's gen2, then we don't need to resolve it again because it won't move" trick with the expr JIT.

Reply:
In general, with a special node `objref` or `const_obj` or somesuch, that should be added by the expr tree builder. And then - to top it off - we can make that a GC root that is automatically updated.